File: leetcode/009-palindrome-number/main.ts
function isPalindrome(x: number): boolean {
	if(x < 0) {
		return false
	}
	let aux: number = x
	let digit: number = 0
	let reversed: number = 0
	while(aux != 0){
		digit = aux % 10
		aux = Math.floor(aux / 10)
		reversed = reversed * 10 + digit
	}
	if(x - reversed === 0) {
		return true
	}
	return false
}

console.log(isPalindrome(121))
console.log(isPalindrome(-121))
console.log(isPalindrome(10))

File: leetcode/009-palindrome-number/main.cpp
#include <iostream>
#include <climits>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        int original = x, reversed = 0;
        while (x != 0) {
            int digit = x % 10;
            if (reversed > (INT_MAX - digit) / 10) return false;
            reversed = reversed * 10 + digit;
            x /= 10;
        }
        return original == reversed;
    }
};

int main() {
    Solution solution;
    int testNumber = 121;
    cout << "Is " << testNumber << " a palindrome? " << (solution.isPalindrome(testNumber) ? "Yes" : "No") << endl;
    return 0;
}